src/socket/address.cr

```diff
@@ -190,6 +197,23 @@ class Socket
       end
     end
 
+    # Returns `true` if this IP is a loopback address.
+    #
+    # In the IPv4 family, loopback addresses are all addresses in the subnet
+    # `127.0.0.0/24`. In IPv6 `::1` is the loopback address.
+    def loopback? : Bool
+      if family == Family::INET
+        @addr4.not_nil!.s_addr & 0x00000000ff_u32 == 0x0000007f_u32
+      else
+        @address == LOOPBACK6
```

Comment:
What if the address is `0000:0000:0000:0000:0000:0000:0000:0001` ? It should work too

Reply:
Yeah, true. But maybe it's better to normalize the address...